rclpy/rclpy/parameter_service.py

```diff
@@ -138,12 +145,19 @@ def _set_parameters_callback(self, request, response):
         return response
 
     def _set_parameters_atomically_callback(self, request, response):
+        node = self._get_node()
         try:
-            response.result = self._node.set_parameters_atomically([
+            response.result = node.set_parameters_atomically([
                 Parameter.from_parameter_msg(p) for p in request.parameters])
         except ParameterNotDeclaredException as e:
             response.result = SetParametersResult(
                     successful=False,
                     reason=str(e)
                 )
         return response
+
+    def _get_node(self):
+        node = self._node_weak_ref()
+        if node is None:
+            raise RuntimeError('Expected valid node weak reference')
```

Comment:
Should this be a `ValueError` instead (or what is the reason to choose `RuntimeError`)?

Reply:
AFAIU, [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError) is used when an argument passed by the user is not of the appropiate value. There's not "argument" involved here, just an attribute of the object being invalid.

`RuntimeError` is the choice when the error doesn't fall in any of the other categories ([link](https://docs.python.org/3/library/exceptions.html#RuntimeError)), and I don't think there is a more appropiate standard error.
I also can create a custom error if that sounds better.

Reply:
I was mostly wondering because `RuntimeError` is treated differently by the command line tools - they suppress the stacktrace and only show the exception string. Not sure if that is intended in this case.

Reply:
It sounds like [ReferenceError](https://docs.python.org/3/library/exceptions.html#ReferenceError) is what we want here.

Reply:
> It sounds like ReferenceError is what we want here.

Sounds like the right option! Updated the PR.

> I was mostly wondering because RuntimeError is treated differently by the command line tools - they suppress the stacktrace and only show the exception string. Not sure if that is intended in this case.

A little off-topic:
I have observed this, and find that behavior strange. Why do the command line tools do that?

Reply:
Some kind of error messages don't warrant to show a stacktrace to the user. And in those cases we use `RuntimeError`.